client/state/selectors/can-current-user-start-site-owner-transfer.ts

```diff
@@ -38,5 +40,15 @@ export default function canCurrentUserStartSiteOwnerTransfer(
 		return false;
 	}
 
+	const inviteCacheKey = 'accepted_site_transfer_invite';
+	const acceptedInvite = store.get( inviteCacheKey );
+
+	store.remove( inviteCacheKey );
+
+	// Allow user to accept the site transfer if they just have accepted the site transfer.
+	if ( acceptedInvite && acceptedInvite.sentTo === userEmail ) {
+		return true;
+	}
+
```

Comment:
It's not clear to me why we need to allow users whose `siteOwnerId !== userId;` to be able to start a new site owner transfer. I would expect that if the transfer has been successful then `siteOwnerId === userId;`

Reply:
This is required to allow new users to accept site transfer before we actually make them owner. And, we are using the existing transfer screen to make it happen. So when a new user visits this screen, the condition `siteOwnerId === userId` won't be true anymore.

Reply:
Still, I can't find where `canCurrentUserStartSiteOwnerTransfer` is used in this flow. Can you direct me to where it's used?

Reply:
Is this related to the next screen where the transfer happens?

Reply:
I have removed the code which isn't part of this diff screen. I will add them in follow up PR for confirmation screen layout changes.